### ISSUE TITLE
iOS SSL Script issue 

### ISSUE THREAD

**Issue by phongleq** [FreeRDP/FreeRDP#2651]
So I've notice that when using the scripts to generate to do the SSL part on iOS, that after building for i386 things something weird is happening. It looks like after copying libcrypto.a and libssl.a to libssl_i386.a and the other file, both files are then removed from their original folder. This prevents from the others being made. So in my instance, it fail to generate one for armv7 and armv7s because the file was not longer there to be copied. 

I'm able to get around this because I have a older version that had generate them correctly, but it looks like this new master script is off. Im not proficient in scripting to know how to fix it.

**Comment by lemonmojo** [comment]
+1

**Comment by bmiklautz** [comment]
@phongleq you are referring to scripts/OpenSSL-DownloadAndBuild.command ?

**Comment by phongleq** [comment]
Yes

**Comment by bmiklautz** [comment]
@phongleq after the script run successfully did you check the folder external/openssl/lib/? In there you should fine two Mach-O universal binaries (libssl.a and libcrypto.a) which contain *all* built architectures. I just tested on our build and test machine. If this doesn't work for you can you please add a more detailed information on the exact error you are getting ("fail to generate one for armv7.." isn't a very exact error description ;()

**Comment by phongleq** [comment]
There are two binaries there, but after i create the Xcode project Xcode complains that the openssl is not build for armv7 or armv7s. I get this when building the script:

```
Building architecture armv7. Please wait ...
Done. Build log saved in BuildLog.darwin-armv7.txt
cp: libcrypto.a: No such file or directory
cp: libssl.a: No such file or directory

Building architecture armv7s. Please wait ...
Done. Build log saved in BuildLog.darwin-armv7s.txt
cp: libcrypto.a: No such file or directory
cp: libssl.a: No such file or directory
```

So im guessing they were not added to the universal binary. So I looked into the logs and it seems the logs had this as the error:

```clang: error: invalid version number in '-miphoneos-version-min=.sd'```

I was previously able to do this last year with Xcode 5 and iOS7. So I didn't think there was a minimum specs needed.

**Comment by bmiklautz** [comment]
@phongleq can you test if the branch ios/openssl in my repository fixes your problem? -https://github.com/bmiklautz/FreeRDP/tree/ios/openssl

**Comment by DonMag** [comment]
@bmiklautz - I just did a quick test, as I too have been using OpenSSL libs built from another source... The OpenSSL-DownloadAndBuild.command script in your referenced branch does indeed appear to fix this issue, as well as the arm64 issue here: https://github.com/FreeRDP/FreeRDP/issues/2680

- Don

**Comment by bmiklautz** [comment]
@DonMag thank you for the feedback. Will see that this finds it way to master ;)

**Comment by lemonmojo** [comment]
@bmiklautz I too can confirm that your branch fixes the issue.

**Comment by phongleq** [comment]
This worked for me. Just a off note what would the procedure to build iFreeRDP for arm64?

**Comment by bmiklautz** [comment]
@phongleq it's not yet tested and supported (I think there are some issues left). However if you have a look at #2162 you will find more information on that.

**Comment by akallabeth** [comment]
Fixed on master 3c7cc555a5a5f02366918be7eb935bb1f22115ef
